### DISCUSSION TITLE
didn't get good result deepAR negative binomial

### DISCUSSION THREAD

**Discussion by ramdhan1989** [awslabs/gluonts#807] [Q&A]
Kindly need your advice how to get the good model using deepAR neg binomial. I have ts data to predict sales with input data such as time information (day, month, hour), weather condition, and holiday. Unfortunately I only get RMSE above 8 on hold out dataset meanwhile using statistic model I can get RMSE 4-5. the loss is stable at 1.99 (what is the loss calculation here?) . tuned hyperparameter some times give lower loss around 1.4 but the evaluation doesn't make sense (RMSE 300). followed some recommendations here https://github.com/awslabs/gluon-ts/issues/636 didn't improve the model.

thanks

**Comment by fernandocamargoai** [comment]
Have you tried the latest version? There are some fixes for NegativeBinomial.

**Comment by esbraun** [comment]
I'm also having an accuracy issue with DeepAR trained on count data with negative binomial output. Oddly, the behavior is restricted to CPU training. I can train the same model with the same data on a GPU and it's fine, but on a CPU, I get extreme, non nonsensical confidence limits and poor accuracy. Student T happens to work fine on both CPU and GPU.

I'm happy to share technical details / output - what would be most useful for me to share?

And I'd be remiss if I didn't mention that this library is absolutely phenomenal.

**Comment by lostella** [comment]
@ramdhan1989 adding the time information explicitly is not needed, since such features are automatically added by DeepAR.

@esbraun With what version of gluonts are you experiencing the issue? There were some changes to NegativeBinomial recently which may have solved the issue.

In both cases, if something does not seem to be working it's better to open a bug report (instead of a question) with exhaustive description of the environment (gluonts version, mxnet version, OS, ...) and a code snippet clearly reproducing the setting used (estimator, hyperparameters, and possibly some data even if artificial).
